### PR TITLE
Cleanup API Operations

### PR DESCRIPTION
**Pull request by ghostzero**
For a long time I had the idea to outsource the API operations as separate trais. I've finally put that into effect. What do you think of it? This has the advantage that typos can be eliminated.

### CODE REVIEW

File: src/Concerns/Api/ClipsTrait.php [hunk 1]
```diff
@@ -2,11 +2,14 @@
...
 namespace romanzipp\Twitch\Concerns\Api;
 
-use romanzipp\Twitch\Helpers\Paginator;
+use romanzipp\Twitch\Concerns\Operations\GetTrait;
+use romanzipp\Twitch\Concerns\Operations\PostTrait;
 use romanzipp\Twitch\Result;
 
 trait ClipsTrait
 {
+    use GetTrait, PostTrait;
```
Comment (romanzipp) [inline src/Concerns/Api/ClipsTrait.php:11]:
```suggestion
    use GetTrait;
    use PostTrait;
```

File: src/Concerns/Api/ExtensionsTrait.php [hunk 2]
```diff
@@ -2,11 +2,14 @@
...
 namespace romanzipp\Twitch\Concerns\Api;
 
-use romanzipp\Twitch\Helpers\Paginator;
+use romanzipp\Twitch\Concerns\Operations\GetTrait;
+use romanzipp\Twitch\Concerns\Operations\JsonTrait;
 use romanzipp\Twitch\Result;
 
 trait ExtensionsTrait
 {
+    use GetTrait, JsonTrait;
```
Comment (romanzipp) [inline src/Concerns/Api/ExtensionsTrait.php:11]:
```suggestion
    use GetTrait;
    use JsonTrait;
```

File: src/Concerns/Api/GamesTrait.php [hunk 3]
```diff
@@ -3,11 +3,16 @@
...
 use InvalidArgumentException;
+use romanzipp\Twitch\Concerns\Operations\GetTrait;
+use romanzipp\Twitch\Concerns\Operations\PostTrait;
+use romanzipp\Twitch\Concerns\Operations\PutTrait;
 use romanzipp\Twitch\Helpers\Paginator;
 use romanzipp\Twitch\Result;
 
 trait GamesTrait
 {
+    use GetTrait, PostTrait, PutTrait;
```
Comment (romanzipp) [inline src/Concerns/Api/GamesTrait.php:14]:
```suggestion
    use GetTrait;
    use PostTrait;
    use PutTrait;
```

File: src/Concerns/Api/ModerationTrait.php [hunk 4]
```diff
@@ -2,11 +2,15 @@
...
 namespace romanzipp\Twitch\Concerns\Api;
 
+use romanzipp\Twitch\Concerns\Operations\GetTrait;
+use romanzipp\Twitch\Concerns\Operations\PostTrait;
 use romanzipp\Twitch\Helpers\Paginator;
 use romanzipp\Twitch\Result;
 
 trait ModerationTrait
 {
+    use GetTrait, PostTrait;
```
Comment (romanzipp) [inline src/Concerns/Api/ModerationTrait.php:12]:
```suggestion
    use GetTrait;
    use PostTrait;
```

File: src/Concerns/Api/UsersTrait.php [hunk 5]
```diff
@@ -2,11 +2,14 @@
...
 namespace romanzipp\Twitch\Concerns\Api;
 
-use romanzipp\Twitch\Helpers\Paginator;
+use romanzipp\Twitch\Concerns\Operations\GetTrait;
+use romanzipp\Twitch\Concerns\Operations\PutTrait;
 use romanzipp\Twitch\Result;
 
 trait UsersTrait
 {
+    use GetTrait, PutTrait;
```
Comment (romanzipp) [inline src/Concerns/Api/UsersTrait.php:11]:
```suggestion
    use GetTrait;
    use PutTrait;
```

File: src/Concerns/Api/WebhooksTrait.php [hunk 6]
```diff
@@ -2,11 +2,14 @@
...
 namespace romanzipp\Twitch\Concerns\Api;
 
-use romanzipp\Twitch\Helpers\Paginator;
+use romanzipp\Twitch\Concerns\Operations\GetTrait;
+use romanzipp\Twitch\Concerns\Operations\PostTrait;
 use romanzipp\Twitch\Result;
 
 trait WebhooksTrait
 {
+    use GetTrait, PostTrait;
```
Comment (romanzipp) [inline src/Concerns/Api/WebhooksTrait.php:11]:
```suggestion
    use GetTrait;
    use PostTrait;
```

**Comment by romanzipp** [comment]
I really like the changes including the cleaned up import statements in the main Twitch service. I've requested some changes to follow up with [PSR-12](https://www.php-fig.org/psr/psr-12/#42-using-traits) styles. Thanks!

**Comment by ghostzero** [comment]
Github says "Changes requested" but i apply all suggestion from code review :thinking: Did i miss something? 

screenshot-github com-2020 07 08-06_22_00

**Comment by romanzipp** [comment]
It's fine, the latest commit just needed additional approval

**Pull request merged**